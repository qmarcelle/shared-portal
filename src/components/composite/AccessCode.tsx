import Image from 'next/image';
import AlertIcon from '../../../public/assets/alert_gray.svg';
import { Column } from '../foundation/Column';
import { Row } from '../foundation/Row';
import { TextBox } from '../foundation/TextBox';

export type AccessCodeProps = {
  accessCodeData: string;
};
export const AccessCode = ({ accessCodeData }: AccessCodeProps) => {
  return (
    <Column>
      {accessCodeData && (
        <div className="access-code ">
          <TextBox text="Access Code" className="ml-6 mt-3 text-[10px]" />

          <TextBox
            text={accessCodeData}
            className="ml-3  font-bold text-[15px]"
          />
        </div>
      )}
      {!accessCodeData && (
        <Row className="card-neutral h-[44px]">
          <Image
            src={AlertIcon}
            className="h-[19px] w-[19px] mt-3 ml-3"
            alt="alert"
          />
          <TextBox text="Access code could not load." className="mt-3 ml-2" />
        </Row>
      )}
    </Column>
  );
};
