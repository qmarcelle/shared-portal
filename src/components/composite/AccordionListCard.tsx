import { AccordionInfoItem } from '@/models/accordionInfoItem';
import { IComponent } from '../IComponent';
import { Accordion } from '../foundation/Accordion';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Divider } from '../foundation/Divider';
import { Header } from '../foundation/Header';
import { Spacer } from '../foundation/Spacer';

interface AccordionListCardProps extends IComponent {
  header: string;
  information: AccordionInfoItem[];
}

export const AccordionListCard = ({
  header,
  information,
  className,
}: AccordionListCardProps) => {
  return (
    <Card className={`${className ?? ''} ${header ? 'large-section' : ''}`}>
      <Column className="items-stretch">
        {header && (
          <div>
            <Header className="title-2" text={header}></Header>
            <Spacer size={22} />
          </div>
        )}
        {information.map((info, index) => (
          <>
            <Accordion
              key={index}
              className="px-2 py-4 "
              label={info.title}
              icon={info.icon}
              child={info.body}
              initialOpen={false}
              type="card"
              openIcon={<Image className="pl-2 w-6" src={Down} alt=""></Image>}
              closeIcon={<Image className="pl-2 w-6" src={Up} alt=""></Image>}
              onOpenCallBack={() =>
                info.onOpenCallBack && info.onOpenCallBack()
              }
            ></Accordion>
            {index !== information.length - 1 && <Divider />}
          </>
        ))}
      </Column>
    </Card>
  );
};
