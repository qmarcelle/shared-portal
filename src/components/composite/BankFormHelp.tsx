import { ChatTrigger } from '@/app/clicktochat/components/ChatTrigger';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Header } from '../foundation/Header';
import { RichText } from '../foundation/RichText';
import { Spacer } from '../foundation/Spacer';

export const BankFormHelp = () => (
  <Column>
    <Card type="elevated" className="small-section">
      <Column>
        <Header className="title-2" text="Get Help with the Bank Draft Form" />
        <Spacer size={16} />
        <RichText
          spans={[
            <span key="1">
              If you need help, please reach out to us. You can{' '}
            </span>,
            <span key="2" className="link font-bold">
              <ChatTrigger>start a chat</ChatTrigger>
            </span>,
            <span key="3"> or call us at [1-800-000-000].</span>,
          ]}
        />
      </Column>
    </Card>
  </Column>
);
