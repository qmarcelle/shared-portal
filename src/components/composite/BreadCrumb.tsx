import '@/styles/breadcrumb.css';
import Image from 'next/image';
import Link from 'next/link';
import { rightWhiteIcon } from '../foundation/Icons';
import { IComponent } from '../IComponent';
export interface BreadcrumbItem {
  path: string;
  label: string;
}

export interface BreadcrumbProps extends IComponent {
  items: BreadcrumbItem[];
}

export const BreadCrumb = ({ items }: BreadcrumbProps) => {
  const current = items.length - 2;

  function computePathLink(index: number) {
    return items
      .slice(0, index - (items.length-1))
      .map((item) => item.path)
      .join("");
  }

  return (
    <nav aria-label="Breadcrumb" className="flex items-center">
      <ol className="breadcrumb items-center body-2 ">
        <>
          {items.map((item, index) => (
            <li key={index} className="hidden md:flex items-center ml-0">
              {index === items.length - 1 ? (
                <span aria-current="page" className="breadcrumb-current">
                  {item.label}
                </span>
              ) : (
                <>
                  <Link
                    className="text-primary font-bold"
                    href={computePathLink(index)}
                  >
                    {item.label}
                  </Link>
                  <div className="px-2">
                    <Image
                      src={rightWhiteIcon}
                      alt=""
                      className="mx-1 h-4 w-4"
                    />
                  </div>
                </>
              )}
            </li>
          ))}
        </>
      </ol>
    </nav>
  );
};

