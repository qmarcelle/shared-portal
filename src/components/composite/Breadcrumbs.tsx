'use client';

import { Breadcrumb } from '@/models/app/breadcrumb';
import '@/styles/breadcrumb.css';
import Image from 'next/image';
import Link from 'next/link';
import React from 'react';
import { rightWhiteIcon } from '../foundation/Icons';

export interface BreadcrumbsProps {
  breadcrumbs: Breadcrumb[];
}

const Breadcrumbs: React.FC<BreadcrumbsProps> = ({
  breadcrumbs,
}: BreadcrumbsProps) => {
  return (
    <nav aria-label="Breadcrumb" className="flex items-center">
      <ol className="breadcrumb items-center body-2 ">
        <>
          {breadcrumbs.map((breadcrumb, index) => (
            <li key={index} className="flex items-center ml-0">
              {index === breadcrumbs.length - 1 ? (
                <span aria-current="page" className="breadcrumb-current">
                  {breadcrumb.title}
                </span>
              ) : (
                <>
                  <Link
                    className="text-primary font-bold"
                    href={breadcrumb.path}
                  >
                    {breadcrumb.title}
                  </Link>
                  <div className="px-2">
                    <Image
                      src={rightWhiteIcon}
                      alt=""
                      className="mx-1 h-4 w-4"
                    />
                  </div>
                </>
              )}
            </li>
          ))}
        </>
      </ol>
    </nav>
  );
};

export default Breadcrumbs;
