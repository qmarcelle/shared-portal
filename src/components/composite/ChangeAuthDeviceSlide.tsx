import { googleAnalytics } from '@/utils/analytics';
import { ReactElement } from 'react';
import { Button } from '../foundation/Button';
import { Column } from '../foundation/Column';
import { Header } from '../foundation/Header';
import { Spacer } from '../foundation/Spacer';
import { TextBox } from '../foundation/TextBox';

/* eslint-disable @typescript-eslint/no-explicit-any */
interface ChangeAuthDeviceProps {
  label: string;
  subLabel: string;
  actionArea: ReactElement;
  bottomNote?: string;
  nextCallback: (() => any) | undefined;
  cancelCallback: () => any;
}

export const ChangeAuthDeviceSlide = ({
  label,
  subLabel,
  actionArea,
  bottomNote,
  nextCallback,
  cancelCallback,
}: ChangeAuthDeviceProps) => {
  return (
    <Column className="items-center">
      <Header className="title-2" text={label} />
      <Spacer size={16} />
      <TextBox className="text-center" text={subLabel} />
      <Spacer size={32} />
      <Column className="w-[352px]">
        {actionArea}
        <Spacer size={24} />
        {bottomNote && <TextBox className="mb-8" text={bottomNote} />}
        <Button
          callback={() => {
            nextCallback?.();
            googleAnalytics(
              'next',
              undefined,
              label,
              'next',
              'select_content',
              'select',
            );
          }}
          label="Next"
        />
        <Spacer size={16} />
        <Button
          callback={() => {
            cancelCallback?.();
            googleAnalytics(
              'cancel',
              undefined,
              label,
              'cancel',
              'select_content',
              'select',
            );
          }}
          type="ghost"
          label="Cancel"
        />
      </Column>
    </Column>
  );
};
