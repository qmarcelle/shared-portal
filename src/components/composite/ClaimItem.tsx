import { IComponent } from '@/components/IComponent';
import { Card } from '@/components/foundation/Card';
import { Column } from '@/components/foundation/Column';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { StatusLabel } from '@/components/foundation/StatusLabel';
import { TextBox } from '@/components/foundation/TextBox';
import { ClaimDetails } from '@/models/claim_details';
import { formatCurrency } from '@/utils/currency_formatter';
import Image from 'next/image';
import DentalIcon from '../../../public/assets/dental.svg';
import MedicalIcon from '../../../public/assets/medical.svg';
import PharmacyIcon from '../../../public/assets/pharmacy.svg';
import VisionIcon from '../../../public/assets/vision.svg';
import { Header } from '../foundation/Header';

interface ClaimItemProps extends IComponent {
  // TODO: Update the model and type while integrating the api.
  claimInfo: ClaimDetails;
  callBack?: (id: string) => void;
}

export const ClaimItem = ({
  claimInfo,
  className,
  callBack,
}: ClaimItemProps) => {
  function getSuccessStatus() {
    switch (claimInfo.claimStatus) {
      case 'Processed':
        return 'success';
      case 'Paid':
        return 'success';
      case 'Denied':
        return 'error';
      case 'Pending':
        return 'neutral';
      case 'Partial Approval':
        return 'partialapproval';
      case 'Approved':
        return 'success';
      case 'Completed':
        return 'success';
      default:
        return 'empty';
    }
  }

  function getFormattedDataValue(
    value: number | string | null,
    defaultValue: string,
    isDollar: boolean,
  ): string | undefined {
    if (value) {
      if (isDollar) {
        return formatCurrency(value as number);
      } else {
        // For non-dollar values, format numbers with 2 decimal places without currency symbol
        return typeof value === 'number' ? value.toFixed(2) : (value as string);
      }
    } else {
      return defaultValue;
    }
  }

  function getClaimIcon() {
    if (claimInfo.claimType == 'Medical') {
      return MedicalIcon;
    } else if (claimInfo.claimType == 'Dental') {
      return DentalIcon;
    } else if (claimInfo.claimType == 'Vision') {
      return VisionIcon;
    } else {
      return PharmacyIcon;
    }
  }

  function getClaimItem() {
    return (
      <section className="md:flex md:flex-row align-top m-4">
        <Image
          src={getClaimIcon()}
          className="icon max-md:hidden"
          alt={claimInfo.claimType}
        />
        <Spacer axis="horizontal" size={8} />
        <Column className="min-w-[95%]">
          <section className="md:flex md:flex-row md:justify-between">
            <section className="flex flex-row mb-[10px] md:flex-col">
              <Image
                src={getClaimIcon()}
                className="icon md:hidden"
                alt={claimInfo.claimType}
              />
              <Header
                text={claimInfo.issuer}
                type="title-3"
                className="body-bold primary-color max-md:ml-[5px]"
              />
            </section>
            <section className="flex flex-row justify-between md:flex-col max-lg:mr-[5px]">
              <StatusLabel
                label={claimInfo.claimStatus}
                status={getSuccessStatus()}
              />
              {claimInfo.isMiniCard && (
                <Column className="md:hidden">
                  {claimInfo.columns
                    ?.filter((column) => column.label === 'My Share')
                    .map((column, index) => (
                      <TextBox
                        key={index}
                        className="body-1 font-bold mt-2"
                        text={
                          getFormattedDataValue(
                            column.value,
                            column.defaultValue,
                            column.isDollar ?? false,
                          ) ?? column.defaultValue
                        }
                      />
                    ))}

                  {!claimInfo.columns && (
                    <TextBox
                      className="body-1 font-bold mt-2"
                      text={
                        claimInfo.claimTotal != null
                          ? `$${claimInfo.claimTotal}`
                          : ''
                      }
                    />
                  )}
                </Column>
              )}
            </section>
          </section>
          {claimInfo.isMiniCard && (
            <Row className="justify-between">
              <Column>
                <TextBox
                  className="body-1 mt-2"
                  text={`Visited on ${claimInfo.serviceDate}, For ${claimInfo.memberName}`}
                />
              </Column>

              <Column className="max-md:hidden">
                {claimInfo.columns
                  ?.filter((column) => column.label === 'My Share')
                  .map((column, index) => (
                    <TextBox
                      key={index}
                      className="body-1 font-bold mt-2"
                      text={
                        getFormattedDataValue(
                          column.value,
                          column.defaultValue,
                          column.isDollar ?? false,
                        ) ?? column.defaultValue
                      }
                    />
                  ))}

                {!claimInfo.columns && (
                  <TextBox
                    className="body-1 font-bold mt-2"
                    text={
                      claimInfo.claimTotal != null
                        ? `$${claimInfo.claimTotal}`
                        : ''
                    }
                  />
                )}
              </Column>
            </Row>
          )}
          {claimInfo.columns && !claimInfo.isMiniCard && (
            <Row className="mt-2">
              <Column className="mr-2 flex-grow max-md:hidden">
                <TextBox
                  className="body-1"
                  text={`Visited on ${claimInfo.serviceDate}`}
                />
                <TextBox
                  className="body-1 mt-2"
                  text={`For ${claimInfo.memberName}`}
                />
              </Column>
              <Column className="flex-grow mr-5 md:hidden">
                <TextBox className="body-1" text="Visited on" />
                <TextBox className="body-1" text={claimInfo.serviceDate} />
                <TextBox
                  className="body-1 mt-2"
                  text={`For ${claimInfo.memberName}`}
                />
              </Column>
              {claimInfo.columns.map((item, index) => (
                <Column
                  className={`mx-4 flex-grow ${!item.isVisibleInMobile ? 'max-md:hidden' : ''}`}
                  key={index}
                >
                  <TextBox
                    className="body-1 tertiary-color"
                    text={item.label}
                  />
                  <TextBox
                    className={`body-1 mt-2 ${item.isValueBold ? 'font-bold' : ''}`}
                    text={
                      getFormattedDataValue(
                        item.value,
                        item.defaultValue,
                        item.isDollar ?? false,
                      ) ?? item.defaultValue
                    }
                  />
                </Column>
              ))}
            </Row>
          )}
        </Column>
      </section>
    );
  }

  return (
    <Card
      className={`cursor-pointer ${className}`}
      type="button"
      onClick={() => callBack?.(claimInfo.encryptedClaimId ?? '')}
      onKeyDown={(e) => {
        if (e.key === 'Enter' || e.key === ' ') {
          e.preventDefault();
          callBack?.(claimInfo.encryptedClaimId ?? '');
        }
      }}
      role="button"
      aria-label={`${claimInfo.claimType} claim for ${claimInfo.memberName} from ${claimInfo.issuer} with status ${claimInfo.claimStatus}`}
      tabIndex={0}
    >
      {getClaimItem()}
    </Card>
  );
};
