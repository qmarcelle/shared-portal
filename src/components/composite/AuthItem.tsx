import Image from 'next/image';
import AlertIcon from '/assets/alert_gray.svg';
import DentalIcon from '/assets/dental.svg';
import MedicalIcon from '/assets/medical.svg';
import PharmacyIcon from '/assets/pharmacy.svg';
import VisionIcon from '/assets/vision.svg';
import { PriorAuthorizationHelp } from '../composite/PriorAuthorizationHelp';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Divider } from '../foundation/Divider';
import { Header } from '../foundation/Header';
import { Row } from '../foundation/Row';
import { Spacer } from '../foundation/Spacer';
import { StatusLabel } from '../foundation/StatusLabel';
import { TextBox } from '../foundation/TextBox';
import { IComponent } from '../IComponent';

interface priorAuthDetailProps extends IComponent {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
}

export const AuthItem = ({ authInfo }: priorAuthDetailProps) => {
  function getSuccessStatus() {
    switch (authInfo.priorAuthDetailStatus) {
      case 'Processed':
        return 'success';
      case 'Denied':
        return 'error';
      case 'Pending':
        return 'neutral';
      case 'PartialApproval':
        return 'partialapproval';
      case 'Approved':
        return 'success';
      default:
        return 'empty';
    }
  }

  function getClaimIcon() {
    if (authInfo.priorAuthDetailType == 'Medical') {
      return MedicalIcon;
    } else if (authInfo.priorAuthDetailType == 'Dental') {
      return DentalIcon;
    } else if (authInfo.priorAuthDetailType == 'Vision') {
      return VisionIcon;
    } else {
      return PharmacyIcon;
    }
  }

  return (
    <main className="flex flex-col justify-center items-center page priorAuthDetail">
      <Spacer size={32}></Spacer>
      <Column className={'app-content app-base-font-color m-5'}>
        <Row className="md:max-my-8">
          <Image
            src={getClaimIcon()}
            className="icon !w-10 !h-10 w-fit"
            alt={authInfo.priorAuthDetailType}
          />
          <Spacer axis="horizontal" size={8} />
          <Header
            type="title-3"
            text={authInfo.priorAuthDetailName}
            className="md:text-3xl !font-light ml-5 w-3/5 mb-4"
          />
        </Row>
        <Column className="statusDetail">
          <StatusLabel
            label={authInfo.priorAuthDetailStatus}
            status={getSuccessStatus()}
          />
          <TextBox
            type="body-1"
            className="mt-4"
            text={'Visited on ' + authInfo.dateOfVisit}
          ></TextBox>
          <TextBox
            type="body-1"
            className="mt-4"
            text={'For ' + authInfo.member}
          ></TextBox>
          <TextBox
            type="body-1"
            className="mt-4"
            text={'Reference ID : ' + authInfo.PriorAuthReferenceId}
          ></TextBox>
        </Column>
      </Column>
      <Column className="app-content app-base-font-color">
        <section className="flex flex-row items-start app-body">
          <Column className="flex-grow page-section-63_33 items-stretch">
            {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
              authInfo.priorAuthDetailStatus === 'Denied') && (
              <Card className="neutral container mb-4">
                <Column className="flex flex-row align-top m-4">
                  <Row>
                    <Image src={AlertIcon} className="icon" alt="alert" />
                    <Spacer axis="horizontal" size={8} />
                    <TextBox
                      type="body-1"
                      className="font-bold"
                      text="Get Help Understanding Partial Approvals."
                    ></TextBox>
                  </Row>

                  <Row className="flex flex-col flex-grow mt-4">
                    <TextBox
                      type="body-1"
                      text="We've been able to approve part of this request. We will
                      send you a letter explaining why and details on how to ask
                      for an appeal."
                    ></TextBox>
                    <TextBox
                      type="body-1"
                      className="mt-4"
                      text="For more information,Please start to chart or call
                      [1-800-000-000]."
                    ></TextBox>
                  </Row>
                </Column>
              </Card>
            )}
            <Card className="mb-8 p-8">
              <Column>
                <Header type="title-2" text="Service Details" />
                <Spacer size={16} />
                <Column className="mb-8 md:flex-row">
                  <Column className="mr-7">
                    <Header type="title-3" text="Referred By:" />
                    {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
                      authInfo.priorAuthDetailStatus === 'Approved') && (
                      <Card className="neutral container mt-4">
                        <Row className="flex flex-row align-top m-4">
                          <TextBox
                            className="w-4/5"
                            type="body-1"
                            text="Provider Information Unavailable."
                          ></TextBox>
                        </Row>
                      </Card>
                    )}
                    {(authInfo.priorAuthDetailStatus === 'Denied' ||
                      authInfo.priorAuthDetailStatus === 'Pending') && (
                      <Column className="w-52">
                        <TextBox
                          type="body-1"
                          className="font-bold mt-4"
                          text={authInfo.referredName}
                        ></TextBox>
                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Facility Address"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="123 Street Address Road City Town, TN 12345"
                        ></TextBox>

                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Phone"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="(123) 456-7890"
                        ></TextBox>
                      </Column>
                    )}
                  </Column>
                  <Column>
                    <Header type="title-3" text="Referred To:" />
                    {(authInfo.priorAuthDetailStatus === 'Denied' ||
                      authInfo.priorAuthDetailStatus === 'Approved') && (
                      <Card className="neutral container mt-4">
                        <Row className="flex flex-row align-top m-4">
                          <TextBox
                            type="body-1"
                            className="w-4/5"
                            text="Provider Information Unavailable."
                          ></TextBox>
                        </Row>
                      </Card>
                    )}
                    {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
                      authInfo.priorAuthDetailStatus === 'Pending') && (
                      <Column className="w-52">
                        <TextBox
                          type="body-1"
                          className="font-bold mt-4"
                          text={authInfo.referredName}
                        ></TextBox>
                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Facility Address"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="123 Street Address Road City Town, TN 12345"
                        ></TextBox>

                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Phone"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="(123) 456-7890"
                        ></TextBox>
                      </Column>
                    )}
                  </Column>
                </Column>
                <Divider />
                <Row>
                  {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
                    authInfo.priorAuthDetailStatus === 'Approved' ||
                    authInfo.priorAuthDetailStatus === 'Denied') && (
                    <Column className="w-52">
                      <Header
                        type="title-3"
                        text="Provider Facility"
                        className="my-8"
                      />
                      <TextBox
                        type="body-1"
                        className="font-bold"
                        text="Local Care Hospital"
                      ></TextBox>
                      <TextBox
                        type="body-2"
                        className="mt-4"
                        text="Facility Address"
                      ></TextBox>
                      <TextBox
                        type="body-1"
                        className="mt-4"
                        text="123 Street Address Road City Town, TN 12345"
                      ></TextBox>

                      <TextBox
                        type="body-2"
                        className="mt-4"
                        text="Phone"
                      ></TextBox>
                      <TextBox
                        type="body-1"
                        className="mt-4"
                        text="(123) 456-7890"
                      ></TextBox>
                    </Column>
                  )}
                  <Column>
                    {authInfo.priorAuthDetailStatus === 'Pending' && (
                      <Card className="neutral container mt-4">
                        <Row className="flex flex-row align-top m-4">
                          <TextBox
                            className="w-4/5"
                            type="body-1"
                            text="Provider Information Unavailable."
                          ></TextBox>
                        </Row>
                      </Card>
                    )}
                  </Column>
                </Row>
              </Column>
            </Card>
          </Column>
          <Column className=" flex-grow page-section-36_67 items-stretch md:mt-0">
            <PriorAuthorizationHelp />
          </Column>
        </section>
        <Spacer size={32}></Spacer>
      </Column>
    </main>
  );
};
