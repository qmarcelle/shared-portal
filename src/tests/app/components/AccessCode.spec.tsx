import { AccessCode } from '@/components/composite/AccessCode';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = (accessCode: string) => {
  return render(
    <>
      <AccessCode accessCodeData={accessCode} />
    </>,
  );
};

describe('CareTN Access Code Computation', () => {
  it('should render AccessCode correctly', () => {
    const component = renderUI('btnbluechat');
    screen.getByText('Access Code');
    screen.getByText('btnbluechat');
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render Error Scenario correctly', () => {
    const component = renderUI('');
    screen.getByText('Access code could not load.');
    expect(component.baseElement).toMatchSnapshot();
  });
});
