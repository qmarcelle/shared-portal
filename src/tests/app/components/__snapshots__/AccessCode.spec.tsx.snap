// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareTN Access Code Computation should render AccessCode correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="access-code "
      >
        <p
          class="body-1 ml-6 mt-3 text-[10px]"
          style="display: block; max-width: 100%;"
        >
          Access Code
        </p>
        <p
          class="body-1 ml-3  font-bold text-[15px]"
          style="display: block; max-width: 100%;"
        >
          btnbluechat
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`CareTN Access Code Computation should render Error Scenario correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-row card-neutral h-[44px]"
      >
        <img
          alt=""
          class="h-[19px] w-[19px] mt-3 ml-3"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <p
          class="body-1 mt-3 ml-2"
          style="display: block; max-width: 100%;"
        >
          Access code could not load.
        </p>
      </div>
    </div>
  </div>
</body>
`;
